api/v1/categories/PkpCategoryCategoryController.php

```diff
@@ -0,0 +1,409 @@
+ *
+ * @class PkpCategoryCategoryController
+ *
+ * @ingroup api_v1_category
+ *
+ * @brief Handle API requests for category operations.
+ */
+
+namespace PKP\API\v1\categories;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\file\PublicFileManager;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Route;
+use PKP\category\Category;
+use PKP\components\forms\context\CategoryForm;
+use PKP\context\Context;
+use PKP\core\Core;
+use PKP\core\PKPBaseController;
+use PKP\core\PKPRequest;
+use PKP\file\TemporaryFile;
+use PKP\file\TemporaryFileManager;
+use PKP\security\authorization\CanAccessSettingsPolicy;
+use PKP\security\authorization\ContextAccessPolicy;
+use PKP\security\Role;
+
+class PkpCategoryCategoryController extends PKPBaseController
+{
+    /**
+     * @inheritDoc
+     */
+    public function authorize(PKPRequest $request, array &$args, array $roleAssignments): bool
+    {
+        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+        $this->addPolicy(new CanAccessSettingsPolicy());
+
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getHandlerPath(): string
+    {
+        return 'categories';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getRouteGroupMiddleware(): array
+    {
+        return [
+            'has.user',
+            'has.context',
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getGroupRoutes(): void
+    {
+        Route::middleware([
+            self::roleAuthorizer([
+                Role::ROLE_ID_MANAGER,
+                Role::ROLE_ID_SITE_ADMIN,
+            ]),
+        ])->group(function () {
+            Route::get('', $this->getMany(...));
+            Route::get('categoryFormComponent', $this->getCategoryFormComponent(...));
+            Route::post('', $this->add(...));
+            Route::put('saveOrder', $this->saveOrder(...));
+            Route::put('{categoryId}', $this->edit(...))
+                ->whereNumber('categoryId');
+            Route::delete('{categoryId}', $this->delete(...))
+                ->whereNumber('categoryId');
+        });
+    }
+
+    /**
+     * Saves the ordering of categories.
+     */
+    public function saveOrder(Request $illuminateRequest, Context $context): JsonResponse
+    {
+        $context = $this->getRequest()->getContext();
+        $sortedCategories = $illuminateRequest->all();
+        $categoryIds = array_map(function ($category) {return $category['id'];}, $sortedCategories);
+
+        $categoriesFound = Repo::category()->getCollector()
+            ->filterByContextIds([$context->getId()])
+            ->filterByIds($categoryIds)
+            ->getMany();
+
+        $categoriesFoundIds = $categoriesFound->map(fn ($category) => $category->getId())->all();
```

Comment:
Not sure if your code editor is understanding that `$category` is a `Category`, if not, I'm all for typing at least the arguments.